### DISCUSSION TITLE
Kiali OpenID Windows AD authentication discussion

### DISCUSSION THREAD

**Discussion by dirkckau** [kiali/kiali#4370] [Q&A]
Hi, 

I would like to have a discussion about the Kiali LDAP AD authentication. 

I am using Kiali v1.39 in the k8s environment, and I would like to integrate the authentication with the Windows AD domain in the data center.  
I've been working on it for long, but I still couldn't reach my goal. 

Is it an option to use Dex (LDAP connector to Windows AD) and Oauth2 Proxy (Collect user input)? 

Where is the entry point? 
1) Kiali -> OAuth2 Proxy -> Dex -> Kiali? 
2) OAuth2 Proxy -> Dex -> Kiali?

It would be amazing if you can share your successful case (work flow) to me. 
All the info that I can searched from Internet are all in pieces, I am not sure if I am working on a correct solution. 

Thanks in advance!

**Comment by israel-hdez** [comment]
LDAP is not supported in Kiali, but as you may have found on the Internet, you can use LDAP with Kiali through an OpenID server that supports LDAP.

I haven't done that setup, but the steps I would follow are:

1. Install Dex and setup LDAP as mentioned in their official documentation: https://dexidp.io/docs/connectors/ldap/
2. Configure a client for Kiali in Dex. See [Dex's example config file for one way to configure](https://github.com/dexidp/dex/blob/3456c3315dd216e53163382a03b01ef0028b4f1d/examples/config-dev.yaml#L105-L118).
3. Configure Kiali to use Dex. Read Kiali docs to learn how: https://kiali.io/documentation/latest/configuration/authentication/openid/

For a user, they would go first to Kiali, which will offer the option to login via "OpenID". Users will be redirected to Dex to login. Once they login successfully, users will be redirected back to Kiali.

You don't need OAuth2 Proxy unless you need it to enable RBAC. It is only needed if you cannot configure OpenID integration in your Kubernetes cluster.
